src/python/pants/base/exception_sink_integration_test.py

```diff
@@ -74,74 +74,72 @@ def assert_graceful_signal_log_matches(pid: int, signum, signame, contents: str)
     assert re.search(regex_str, contents)
 
 
-def test_logs_unhandled_exception() -> None:
+def test_logs_unhandled_exception(tmp_path: Path) -> None:
     directory = "testprojects/src/python/hello/main"
-    with temporary_dir() as tmpdir:
-        pants_run = run_pants_with_workdir(
-            [
-                "--no-pantsd",
-                "list",
-                f"{directory}:this-target-does-not-exist",
-                "--backend-packages=['pants.backend.python']",
-            ],
-            workdir=tmpdir,
-            # The backtrace should be omitted when --print-stacktrace=False.
-            print_stacktrace=False,
-            hermetic=False,
-        )
 
-        pants_run.assert_failure()
+    pants_run = run_pants_with_workdir(
+        [
+            "--no-pantsd",
+            "list",
+            f"{directory}:this-target-does-not-exist",
+            "--backend-packages=['pants.backend.python']",
+        ],
+        workdir=tmp_path.as_posix(),
```

Comment:
This is fine in practice and it's done often enough in the Pants codebase, but I imagine  `str(path)` is more correct (stands a chance of working on Windows if / when we get there).